rpcserver.go

```diff
@@ -4657,6 +4663,43 @@ func marshalDbEdge(edgeInfo *channeldb.ChannelEdgeInfo,
 	return edge
 }
 
+// GetNodeMetrics returns all available node metrics calculated from the
+// current channel graph.
+func (r *rpcServer) GetNodeMetrics(ctx context.Context,
+	req *lnrpc.NodeMetricsRequest) (*lnrpc.NodeMetricsResponse, error) {
+
+	resp := &lnrpc.NodeMetricsResponse{}
+
+	// Obtain the pointer to the global singleton channel graph, this will
+	// provide a consistent view of the graph due to bolt db's
+	// transactional model.
+	graph := r.server.chanDB.ChannelGraph()
+
+	// Calculate betweenness centrality if requested. Note that depending on the
+	// graph size, this may take up to a few minutes.
+	channelGraph := autopilot.ChannelGraphFromDatabase(graph)
+	centralityMetric := autopilot.NewBetweennessCentralityMetric()
+	if err := centralityMetric.Refresh(channelGraph); err != nil {
+		return nil, err
+	}
+	resp.Metrics = make(map[string]*lnrpc.NodeMetrics)
+
+	// Fill normalized and non normalized centrality.
+	centrality := centralityMetric.GetMetric(true)
+	for nodeID, val := range centrality {
+		resp.Metrics[hex.EncodeToString(nodeID[:])] = &lnrpc.NodeMetrics{
```

Comment:
i believe we can cast these directly as `string(nodeID[:])`, they will still be printed properly in lncli due to hex-protobuf-display

Reply:
Unfortunately this is not the case, hex encoding is needed.

`Internal desc = grpc: error while marshaling: proto: field "lnrpc.NodeMetricsResponse.Metrics" contains invalid UTF-8`

Reply:
bummer, strings it is!